File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/36_삽입정렬/main.cpp
#include <stdio.h>

int main() {
	int a[100], i, j, n, tmp;

	scanf("%d", &n);

	for (i = 0; i < n; i++) {
		scanf("%d", &a[i]);
    }

    for (i = 0; i < n; i++) {
		tmp = a[i];
		for(j = i-1; j >= 0; j--) {
			if (a[j] > tmp) {
				a[j+1] = a[j];
            } else {
				break;
			}
		}
		a[j+1] = tmp;
	}
	
	for (i=0; i< n; i++) {
		printf("%d ", a[i]);
    }

	return 0;
}

File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/36_삽입정렬/main.go
package main

import "fmt"

func main() {
	var N int
	var A [100]int
	fmt.Scanf("%d", &N)

	for i := 0; i < N; i++ {
		fmt.Scanf("%d", &A[i])
	}

	for i := 1; i < N; i++ {
		tmp := A[i]
		index := i - 1
		for index >= 0 && A[index] > tmp {
			A[index+1] = A[index]
			index--
		}
		A[index+1] = tmp
	}
}
